pkg/services/ngalert/eval/eval.go

```diff
@@ -155,11 +162,57 @@ func executeCondition(ctx AlertExecCtx, c *models.Condition, now time.Time, data
 		return ExecutionResults{Error: err}
 	}
 
+	// eval captures for the '__value__' label.
+	captures := []NumberValueCapture{}
+
+	captureVal := func(refID string, labels data.Labels, value *float64) {
+		captures = append(captures, NumberValueCapture{
+			Var:    refID,
+			Value:  value,
+			Labels: labels.Copy(),
+		})
+	}
+
 	for refID, res := range execResp.Responses {
-		if refID != c.Condition {
-			continue
+		// for each frame within each response, capture value if it is a number.
+		for _, frame := range res.Frames {
+			if len(frame.Fields) != 1 || frame.Fields[0].Type() != data.FieldTypeNullableFloat64 {
+				continue
+			}
+			var v *float64
+			if frame.Fields[0].Len() == 1 {
+				v = frame.At(0, 0).(*float64) // type checked above
+			}
+			captureVal(frame.RefID, frame.Fields[0].Labels, v)
+		}
+
+		if refID == c.Condition {
+			result.Results = res.Frames
+		}
+	}
+
+	// add capture values as data frame metadata to each result (frame) that has matching labels.
+	for _, frame := range result.Results {
+		if frame.Meta != nil && frame.Meta.Custom != nil {
```

Comment:
```suggestion
        // classic conditions already have metadata set and only have one value, there's no need to add anything in this case.
		if frame.Meta != nil && frame.Meta.Custom != nil {
```
Is this understanding correct? If so, I'd replace the comment below with this one.